chainerx_cc/chainerx/native/native_device/linalg.cc

```diff
@@ -360,5 +390,70 @@ class NativeSvdKernel : public SvdKernel {
 
 CHAINERX_NATIVE_REGISTER_KERNEL(SvdKernel, NativeSvdKernel);
 
+class NativeSyevdKernel : public SyevdKernel {
+public:
+    void Call(const Array& a, const Array& w, const Array& v, char uplo, bool compute_v) override {
+#if CHAINERX_ENABLE_LAPACK
+        Device& device = a.device();
+        Dtype dtype = a.dtype();
+
+        CHAINERX_ASSERT(a.ndim() == 2);
+
+        // Syevd stores the result in-place, copy a to v to avoid destroying the input matrix
+        device.backend().CallKernel<CopyKernel>(a, v);
+
+        int64_t m = a.shape()[0];
+        int64_t n = a.shape()[1];
+
+        auto syevd_impl = [&](auto pt) {
+            using T = typename decltype(pt)::type;
+
+            auto v_ptr = static_cast<T*>(internal::GetRawOffsetData(v));
+            auto w_ptr = static_cast<T*>(internal::GetRawOffsetData(w));
+
+            char jobz = compute_v ? 'V' : 'N';
+
+            // LAPACK assumes that arrays are stored in column-major order
+            // The uplo argument is swapped instead of transposing the input matrix
+            char uplo_swapped = (uplo == 'U') ? 'L' : 'U';
+
+            int info;
+            int lwork = -1;
+            int liwork = -1;
+            T work_size;
+            int iwork_size;
+
+            // When calling Syevd matrix dimensions are swapped instead of transposing the input matrix
+            Syevd<T>(jobz, uplo_swapped, n, v_ptr, m, w_ptr, &work_size, lwork, &iwork_size, liwork, &info);
+
```

Comment:
Should you check `info` before proceeding to the next LOC?

Reply:
The first call is a workspace query. The routine only calculates the required sizes of workspace arrays, returns these values, and no error message is issued.
src: https://software.intel.com/en-us/mkl-developer-reference-fortran-syevd

Reply:
As written in docs this step is not necessary if we are willing to provide plenty of memory for workspace, but the execution
>though probably not so fast as with a recommended workspace.